### ISSUE TITLE
reassign issues false positives with shadowed names

### ISSUE THREAD

**Issue by MichaelUrman** [golangci/golangci-lint#3145]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

When running `golangci-lint run -E reassign` on the reproduction below, it reports an error for writing to a local variable whose name shadows an imported package. For example. `...: reassigning variable EOF in other package fmt`. (As we know there is no EOF in the fmt package.)

This is a simplification from where I first encountered it. There the shadowed name was a more realistic variable name, and I had configured patterns to include `".*"`.

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.49.0 built from cc2d97f3 on 2022-08-24T10:24:37Z
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
cat: .golangci.yml: No such file or directory
```

</details>

### Go environment

<details>

```console
$ go version && go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/murman/.cache/go-build"
GOENV="/home/murman/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/murman/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/murman/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/home/murman/sdk/go1.18.3"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/home/murman/sdk/go1.18.3/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.18.3"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/murman/src/reassign-broke/go.mod"
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build2937758740=/tmp/go-build -gno-record-gcc-switches"

```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v -E reassign
level=info msg="[config_reader] Config search paths: [./ /home/murman/src/reassign-broke /home/murman/src /home/murman /home /]"
level=info msg="[lintersdb] Active 8 linters: [errcheck gosimple govet ineffassign reassign staticcheck typecheck unused]"
level=info msg="[loader] Go packages loading at mode 575 (types_sizes|deps|files|imports|compiled_files|exports_file|name) took 58.290525ms"
level=info msg="[runner/filename_unadjuster] Pre-built 0 adjustments in 98.187µs"
level=info msg="[linters context/goanalysis] analyzers took 564.491422ms with top 10 stages: buildir: 448.630554ms, inspect: 24.007578ms, ctrlflow: 18.78098ms, printf: 16.826543ms, fact_deprecated: 12.915033ms, SA5012: 12.324921ms, fact_purity: 8.920857ms, nilness: 8.209989ms, typedness: 7.312996ms, SA1005: 369.346µs"
level=info msg="[runner] Processors filtering stat (out/in): sort_results: 1/1, filename_unadjuster: 1/1, skip_dirs: 1/1, exclude-rules: 1/1, path_shortener: 1/1, severity-rules: 1/1, path_prefixer: 1/1, cgo: 1/1, path_prettifier: 1/1, autogenerated_exclude: 1/1, exclude: 1/1, max_from_linter: 1/1, source_code: 1/1, identifier_marker: 1/1, nolint: 1/1, uniq_by_line: 1/1, max_per_file_from_linter: 1/1, skip_files: 1/1, diff: 1/1, max_same_issues: 1/1"
level=info msg="[runner] processing took 128.689µs with stages: nolint: 48.637µs, identifier_marker: 25.955µs, exclude-rules: 17.598µs, autogenerated_exclude: 11.767µs, path_prettifier: 9.143µs, source_code: 5.072µs, skip_dirs: 2.778µs, uniq_by_line: 2.423µs, max_same_issues: 942ns, cgo: 813ns, filename_unadjuster: 780ns, path_shortener: 575ns, exclude: 541ns, max_from_linter: 481ns, diff: 306ns, skip_files: 270ns, max_per_file_from_linter: 255ns, severity-rules: 177ns, sort_results: 104ns, path_prefixer: 72ns"
level=info msg="[runner] linters took 537.834154ms with stages: goanalysis_metalinter: 537.655051ms"
foo.go:9:5: reassigning variable EOF in other package fmt (reassign)
    fmt.EOF = 5
    ^
level=info msg="File cache stats: 1 entries of total size 114B"
level=info msg="Memory: 8 samples, avg is 145.0MB, max is 212.3MB"
level=info msg="Execution took 605.210369ms"

```

</details>

### Code example or link to a public repository

<details>

```go
package main

import "fmt"

func main() {
    fmt.Println("hi")

    fmt := struct{ EOF int }{}
    fmt.EOF = 5
}
```

</details>

**Comment by ldez** [comment]
Hello,

it's an issue with [reassign] so you have to open an issue [here](https://github.com/curioswitch/go-reassign/issues)

**Comment by MichaelUrman** [comment]
Thanks. I thought you'd want to know of the false positive, given the requirements on new linters. I'll follow up there.
